src/sys/ptrace.rs

```diff
@@ -280,3 +280,30 @@ pub fn cont<T: Into<Option<Signal>>>(pid: Pid, sig: T) -> Result<()> {
     }
 }
 
+/// Move the stopped tracee process forward by a single step as with 
+/// `ptrace(PTRACE_SINGLESTEP, ...)`
+///
+/// Advances the execution of the process with PID `pid` by a single step optionally delivering a
+/// signal specified by `sig`.
+///
+/// # Example
+/// ```rust
+/// extern crate nix;
+/// use nix::sys::ptrace::step;
+/// use nix::unistd::Pid;
+/// use nix::sys::signal::Signal; 
+/// fn main() {
+///     let dummy_pid = Pid::from_raw(0); 
+///
+///     let _ = step(dummy_pid, Some(Signal::SIGSTOP));
```

Comment:
Shouldn't you not need that `Some()` on this line?

Also, any way to make this a more realistic example? This looks more like a basic test than any realistic code that might help a user use this function.